managed/devops/bin/yb_platform_util_py2.py

```diff
@@ -0,0 +1,498 @@
+            return func(*args, **kwargs)
+        except urllib2.HTTPError as e:
+            content = e.read().decode("utf-8")
+            if "html>" in content:
+                message = "Invalid YB_PLATFORM_URL URL or params, Getting html page in response"
+                response = {"data": message, "status": "failed", "error": message}
+                print(response)
+            else:
+                response = {"data": content, "status": "failed", "error": content}
+                print(response)
+        except Exception as e:
+            response = {"data": str(e), "status": "failed", "error": str(e)}
+            print(response)
+    return inner_function
+
+
+def convert_unicode_json(data):
+    """
+    Function to convert unicode json to dictionary
+    {u"name": u"universe"} => {"name": "universe"}
+    """
+    if isinstance(data, basestring):
+        return str(data)
+    elif isinstance(data, collections.Mapping):
+        return dict(map(convert_unicode_json, data.iteritems()))
+    elif isinstance(data, collections.Iterable):
+        return type(data)(map(convert_unicode_json, data))
+    else:
+        return data
+
+
+def call_api(url, auth_uuid, data=None, is_delete=False):
+    """
+    Call the corresponding url with auth token, headers and returns the response.
+
+    :param url: url to be called.
+    :param auth_uuid: authentication token of the customer.
+    :param data: data for POST request.
+    :param is_delete: to identify the delete call.
+    :return: response of the API call.
+    """
+    if not is_delete:
+        request = urllib2.Request(url)
+    else:
+        request = urllib2.Request(url)
+        request.get_method = lambda: 'DELETE'
+
+    request.add_header('X-AUTH-YW-API-TOKEN', auth_uuid)
+    request.add_header('Content-Type', 'application/json; charset=utf-8')
+    if data:
+        response = urllib2.urlopen(request, json.dumps(data).encode('utf-8'))
+    else:
+        response = urllib2.urlopen(request)
+    return response
+
+
+def get_universe_details(base_url, customer_uuid, auth_uuid, universe_name, base_dir):
+    """
+    Get the universe details and store it in a json file after formatting the json.
+
+    :param base_url: base url of the platform back end.
+    :param customer_uuid: UUID of the customer.
+    :param auth_uuid: authentication token for the customer.
+    :param universe_name: Name of the universe to take a json data from.
+    :param base_dir: base directory in which the json file should be stored.
+    :return: None
+    """
+    universe = get_universe_by_name(base_url, customer_uuid, auth_uuid, universe_name)
+    if universe:
+        configure_json = create_universe_config(universe, universe["name"])
+        file = base_dir + '/' + universe["name"] + ".json"
+        with open(file, 'w') as file_obj:
+            json.dump(configure_json, file_obj)
+        response = {
+            "data": "Detail of universe has been saved to " + str(file),
```

Comment:
Details ... have been saved.
Also from the function name it is not clear that it saves something to a file / returns data through a file.